iothub/device/devdoc/API Design.md

```diff
@@ -0,0 +1,255 @@
+## Plug and Play convention compatible APIs
+
+### Properties
+
+```csharp
+/// <summary>
+/// Update a single property.
+/// 
+/// Property under the root component:
+/// "reported": {
+///     "temperature": 21.3
+/// }
+/// 
+/// Property under a specified component:
+/// "reported": {
+///     "thermostat1": {
+///         "__t": "c",
+///         "temperature": 21.3
+///     }
+/// }
+/// </summary>
+/// <param name="propertyName">The name of the twin property to update.</param>
+/// <param name="propertyValue">The value of the twin property to update.</param>
+/// <param name="componentName">The name of the component in which the property is defined. Can be null for property defined under the root interface.</param>
+/// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+public Task UpdatePropertyAsync(string propertyName, dynamic propertyValue, string componentName = default, CancellationToken cancellationToken = default);
+
+/// <summary>
+/// Update a batch of properties.
+/// 
+/// Properties under the root component:
+/// "reported": {
+///     "temperature": 21.3,
+///     "humudity": 60
+/// }
+/// 
+/// Properties under a specified component:
+/// "reported": {
+///     "thermostat1": {
+///         "__t": "c",
+///         "temperature": 21.3,
+///         "humudity": 60
+///     }
+/// }
+/// </summary>
+/// <param name="properties">The twin properties and values to update.</param>
+/// <param name="componentName">The name of the component in which the properties are defined. Can be null for properties defined under the root interface.</param>
```

Comment:
Low prio -- We need to check if this is `root interface` or `root component`.  I think component maybe more correct, but want to call out for later research before I forget.

Reply:
In  IoT Explorer it is named: default component